packages/x-data-grid/src/components/grid/toolbar/GridToolbarButton.tsx

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable react/no-unused-prop-types */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { alpha, styled } from '@mui/material/styles';
+import composeClasses from '@mui/utils/composeClasses';
+import clsx from 'clsx';
+import ButtonBase, { ButtonBaseProps } from '@mui/material/ButtonBase';
+import { useGridRootProps } from '../../../hooks/utils/useGridRootProps';
+import { getDataGridUtilityClass } from '../../../constants/gridClasses';
+import type { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import {
+  useGridComponentRenderer,
+  RenderProp,
+} from '../../../hooks/utils/useGridComponentRenderer';
+
+export interface GridToolbarButtonProps extends ButtonBaseProps {
+  /**
+   * The color of the component.
+   * @default 'standard'
+   */
+  color?: 'standard' | 'primary';
+  /**
+   * A function to customize rendering of the component.
+   */
+  render?: RenderProp<{}>;
+}
+
+type OwnerState = DataGridProcessedProps;
+
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['toolbarButton'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+// Based on @mui/material/ToggleButton styles
+const StyledGridToolbarButton = styled(ButtonBase, {
+  name: 'MuiDataGrid',
+  slot: 'ToolbarButton',
+})<{ ownerState: OwnerState; selected?: boolean }>(({ theme }) => ({
+  ...theme.typography.button,
```

Comment:
Why do we need to do this? It feels like there's something off if we need to redo material styles here.

Also note that you're introducing uses of `theme` that won't be handled by the #15704 PR.

Reply:
This is a regular button, but we want to style it in a similar way to the Material UI Toggle Button. We can't use the Toggle Button because it has different behavior and attributes to a regular button.